### PR TITLE
[ADD] google_meeting module v.7.0

### PR DESCRIPTION
**Pull request by tafaRU**
This PR contains the module **google_meeting** which provides a solution for synchronizing crm_meeting in Odoo v7.0 with Google Calendar.

I applied on the module https://github.com/solutions2use/google_api  developed by @solutions2use  a strong PEP8 cleanup and a bit of refactoring. I also renamed the module to google_meeting because it seemed more suitable.

See the description in the module manifest file for the installation notes.

### CODE REVIEW

**Comment by tafaRU** [comment]
Test server fails in Travis CI builds due to the following error:

 `File "/home/travis/build/OCA/server-tools/google_meeting/__init__.py", line 22, in <module>
    from . import crm_meeting
  File "/home/travis/build/OCA/server-tools/google_meeting/crm_meeting.py", line 24, in <module>
    from apiclient import discovery
ImportError: No module named apiclient`

How can I fix it? Any hints?

**Comment by pedrobaeza** [comment]
`apiclient` it's a Python dependency. Can you add it in Travis file?

**Comment by tafaRU** [comment]
@pedrobaeza thank you, I was just doing ;-)

File: google_meeting/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+#
+##############################################################################
+{
+    'name': 'Google Meeting',
+    'author': 'solutions2use',
+    'website': "http://www.solutions2use.com",
+    'license': 'AGPL-3',
+    'version': "1.0",
+    "depends": [
+        'base_calendar'
```
Comment (OSguard) [inline google_meeting/__openerp__.py:29]: can you add "google-api-python-client" as external_dependecy?

File: google_meeting/google_api_account.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+from oauth2client import file
+from oauth2client import client
+from oauth2client import tools
+
+
+class GoogleApiAccount(orm.Model):
+
+    _name = 'google.api.account'
+    _columns = {
+        'name': fields.char('Account', size=50, required=True),
```
Comment (OSguard) [inline google_meeting/google_api_account.py:35]: is there any need to limit name to 50 chars?

File: google_meeting/google_api_account.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    def do_authorize(self, cr, uid, ids, context=None):
+
+        account = self.browse(cr, uid, ids[0])
+
+        FLOW = client.flow_from_clientsecrets(
+            account.secrets_path,
+            scope=[
+                'https://www.googleapis.com/auth/calendar',
+                'https://www.googleapis.com/auth/calendar.readonly',
+                'https://www.google.com/m8/feeds',
```
Comment (OSguard) [inline google_meeting/google_api_account.py:66]: can we add this scope as static variable at the beginning of the file?

File: google_meeting/google_api_calendar.py [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+            'User already assigned to a google calendar!'
+        ),
+        (
+            'pu-key-2',
+            'UNIQUE (calendar_id)',
+            'Google calendar already assigned to another user!'
+        ),
+    ]
+
+    def do_synchronize(self, cr, uid, ids, context=None):
```
Comment (OSguard) [inline google_meeting/google_api_calendar.py:60]: function is to big, is it possible to split it and use some internal functions?

**Comment by OSguard** [comment]
I help Bas to debug the original module and test it. But i never use it in production. This was only the first hints. I will put on my worklist to make a function test and additional review.

**Comment by tafaRU** [comment]
Thanks a lot @OSguard for your hints. I will apply them asap.

File: google_meeting/google_api_account.py (hunk 2 above)
Comment (tafaRU) [inline google_meeting/google_api_account.py:35]: I don't think, what size do you think it's acceptable?

File: google_meeting/google_api_calendar.py (hunk 4 above)
Comment (tafaRU) [inline google_meeting/google_api_calendar.py:60]: I agree, but for the moment I don't have enough time to do it. What about you? :pray:

File: google_meeting/google_api_account.py (hunk 3 above)
Comment (tafaRU) [inline google_meeting/google_api_account.py:66]: How can I pass account to the global variable? Please, could you provide an example? Thank you in advance.

File: google_meeting/__openerp__.py (hunk 1 above)
Comment (tafaRU) [inline google_meeting/__openerp__.py:29]: See my last commit.

File: google_meeting/google_api_calendar.py (hunk 4 above)
Comment (OSguard) [inline google_meeting/google_api_calendar.py:60]: the same problem, so it would be nice, but no hard requirement

**Comment by tafaRU** [comment]
Please @OSguard could you check all my comments in file diff?
Thank you in advance!

**Comment by eLBati** [comment]
@tafaRU please rebase

**Comment by tafaRU** [comment]
@eLBati just rebased.

**Comment by dreispt** [comment]
Shouldn't this go on the CRM repo?

**Comment by tafaRU** [comment]
@OSguard could you please check if other fixes needed?
Thank you in advance.

**Comment by eLBati** [comment]
@tafaRU 

```
Unable to install module "google_meeting" because an external dependency is not met: No module named google-api-python-client
```

**Comment by tafaRU** [comment]
@eLBati I see, what do you suggest I do?

**Comment by tafaRU** [comment]
Such kind of errors in last build makes me suggest a travis restart could solve them. Someone could please restart it?

**Comment by tafaRU** [comment]
Just rebased so please ignore my last commit. Thanks!

**Comment by tafaRU** [comment]
We don't think to continue with this work, so the PR can be taken from someone else.
Any volunteers?

**Comment by pedrobaeza** [comment]
As no response, I close.

**Pull request closed without merging**